### ISSUE TITLE
Category remove / disable / not visible

### ISSUE THREAD

**Issue by cougil** [alfio-event/alf.io#391]
Hi!
Once we have created a category, it is possible to remove it / make it not visible or disable it? Because I have not seen how to do it, once the category is visible and/or somebody buys a ticket. 
I see the danger of this feature, but it could be really helpful for testing purposes or if some of the admins creates a ticket but later on decides to change his/her mind.
Best,

**Comment by cbellone** [comment]
Hi,

We'll implement the "disable / remove category" functionality in alf.io v2 (exp. Q4 2018). For now, you can hide a category by setting its visibility to "restricted". 

To do that, click on "edit" and:
- set the allocation strategy to "limited seats"
- set the category visibility to "restricted"
- set "max tickets"

and then save. 

Please note that seats modifications are applied after 30 s, in order to allow the waiting queue processor to kick in. The page will refresh automatically.

**Comment by cougil** [comment]
Hi Celestino.
I'd tried to do this just to "deactive" and already visible Category, and basically it is impossible if they are tickets already sold :-/

**Comment by cbellone** [comment]
Hi @cougil,
I see your problem. 
As a workaround, if the ticket(s) have been bought just for test, you can cancel the corresponding reservation (*). 
![image](https://user-images.githubusercontent.com/3385346/36366484-a32c1d66-154e-11e8-95e5-a36e783e6bb6.png)


The ticket will be reverted to FREE after 30s, and that would let you change the category visibility.

(*) if you've done the reservation using the "live" payment provider configuration, you can safely issue a refund, as you're using the live configuration now. Otherwise, please deselect the "Refund xxx" option

**Comment by cougil** [comment]
Ok it worked!
Now even the category does not appear, we can see a warning + and exclamation in the event and in the category warning that there are not sold tickets:
` This category contains orphan (not sold) tickets. Move them to another category:`
Best

**Comment by cbellone** [comment]
OK, I think I've found the problem. Should be caused by #392 .

tl;dr: you've found a bug for a corner case :+1: Thank you!!

Just to prove my analysis, can you confirm that you've received the following email?

Subject: warning: stuck tickets found
Text: there are stuck tickets for the event <your-event-name>. Please check admin area.

The problem is that the "mandatory configuration" error is thrown in a wrong phase, and it wrongly propagate to the payment subsystem. 
For the payment subsystem any unhandled error is an issue happened after receiving a response from the payment gateway (usually when the database connection is lost while updating the status), which could potentially lead to a consistency issue. Therefore the reservation enters the "STUCK" state. 

Additionally, the admin GUI doesn't handle (yet) this case. To cut it short, you ended up having a ticket "PENDING" linked to a "CANCELLED" reservation.

I'll fix this in #392 and include a migration script for solving the issue by resetting the ticket status. 

Thanks again
Celestino

**Comment by cougil** [comment]
Yes, I remember we have received some emails like those ;-)
Cool for the solution, it will be great to avoid this issues!
Btw, because of this, I can imagine that while this issue is not fixed, we will not be able to remove this warning, don't we?
Best,

**Comment by cbellone** [comment]
Yes, I am afraid so. Sorry for that.

...unless you want to fix the issue directly on the database:

first verify that the following query returns exactly one result

```sql
select a.id from ticket a, tickets_reservation b where a.status = 'PENDING' and a.tickets_reservation_id = b.id and b.status = 'CANCELLED';
```
then run the update

```sql
update ticket set status = 'RELEASED', tickets_reservation_id = null, special_price_id_fk = null, 
first_name = null, last_name = null, full_name = null, email_address = null 
where id in (select a.id from ticket a, tickets_reservation b where a.status = 'PENDING' and a.tickets_reservation_id = b.id and b.status = 'CANCELLED');
```

after running the update you should wait ~30s for the waiting queue processor mechanism. After that you can safely move the tickets from the expired category to another one.  

good luck :-)

**Comment by cbellone** [comment]
fixed. Will be released in v1.14
